packages/shared/lib/contexts/onboarding/utils/hasUserClaimedRewards.ts

```diff
@@ -1,14 +1,22 @@
 import { IShimmerClaimingAccount, ShimmerClaimingAccountState } from '@contexts/onboarding'
 
 export function hasUserClaimedRewards(shimmerClaimingAccounts: IShimmerClaimingAccount[]): boolean {
+    let hasAtLeastOneClaimingTransaction = false
     return (
         shimmerClaimingAccounts.every((shimmerClaimingAccount) => {
-            const { state } = shimmerClaimingAccount
-            return (
-                state === ShimmerClaimingAccountState.UnclaimedWithoutRewards ||
-                state === ShimmerClaimingAccountState.FullyClaimed
-            )
-        }) &&
-        shimmerClaimingAccounts?.some((shimmerClaimingAccount) => Boolean(shimmerClaimingAccount?.claimingTransaction))
+            const { state, claimingTransaction } = shimmerClaimingAccount
+            if (state === ShimmerClaimingAccountState.UnclaimedWithoutRewards) {
+                return true
+            } else if (state === ShimmerClaimingAccountState.FullyClaimed && claimingTransaction) {
+                /**
+                 * NOTE: This side effect is here to avoid iterating more than once in
+                 * the case that every account is "unclaimed without rewards".
+                 */
```

Comment:
Because of the fact that having one account unclaimed without rewards (in the code here) is okay unless all of them are unclaimed without rewards. With this extra bit of logic we can just iterate once instead of twice, but not really a big deal since this is just an array of accounts which likely won't be too large anyway.